packages/button/src/ButtonBase.ts

```diff
@@ -215,6 +233,7 @@ export class ButtonBase extends ObserveSlotText(LikeAnchor(Focusable), '', [
         this.manageAnchor();
         this.addEventListener('keydown', this.handleKeydown);
         this.addEventListener('keypress', this.handleKeypress);
+        this.addEventListener('click', this.shouldProxyClick);
```

Comment:
Do we need to add this click event listener? My understanding is that we’re already overriding `click()` in the component. And `click()` handles `shouldProxyClick()` inside of it, so it seems like this event listener might duplicate behavior. Could we rely on the overridden `click()` method alone for this functionality?

Reply:
@rubencarvalho overriding `click()` is only capturing programatic click events, not mouse events, which makes me wonder how many other elements pass our tests with programatic clicks but not with mouse clicks and screen reader clicks.

Refactor will be pushed to this branch momentarily.